### ISSUE TITLE
Add compact formatter

### ISSUE THREAD

**Issue by Ruffeng** [stylelint/stylelint#3415]
# Current problem
So far, there is 3 kind of formatters for `stylelint CLI` : verbose, json and string(default). 
All of them are really nice, but I find the necessity to get an output that it shows file, line, type of error, and description of the error in a very single line.

The current output is something like:
```
some_folder/styles.css
4:1  ⚠ 'Expected ".foo.bar" to have specificity no more than "0,1,0"
6:1  ✖ 'Unexpected unit "px" for property "margin"
7:1  ⚠ 'more warnings..'
```

It would be nice to have another formatter option that does an output like:
```
some_folder/styles.css     4:1  ⚠ 'Expected ".foo.bar" to have specificity no more than "0,1,0"
some_folder/styles.css     6:1  ✖ 'Unexpected unit "px" for property "margin"
some_folder/styles.css     7:1  ⚠ 'more warnings..'
```
## Why to do that?
Can happen that we want to work with the output from the CLI, with this new output, we can easily check line by line which file it belongs to the error.

An example of that is the [Overcommit](https://github.com/brigade/overcommit). To make work the linters, it's based on single line outputs and it does a matching with a regex. Here an issue to implement Stylelint at overcommit [#514](https://github.com/brigade/overcommit/issues/514 )

## Conclusion
In my opinion, it could be another elegant option, kind of `compact` formatter. I saw the formatters files in the project. If I get some guidance, I can try to create a PR with this kind of output ( I assume there is more setup than just create the file in formatters and prepare it).

Looking forward to your thoughts! 

Best!

**Comment by evilebottnawi** [comment]
@Ruffeng like this https://babeljs.io/docs/en/next/babel-code-frame.html?

**Comment by Ruffeng** [comment]
@evilebottnawi not really, I was thinking more just as CLI behavior, and showing exactly the same output as it does now but with the file appended for each single line.

**Comment by Arcanemagus** [comment]
Is using the `--custom-formatter` an option, pointing towards your own formatter that outputs the text in an ideal format for your parsing?

**Comment by Ruffeng** [comment]
Kind of. My scenario, for example, I would really love to append stylelint in the overcommit gem, it means I have no chance to create my own JS file to specify the output I am desiring. The only way it works there is with the output from CLI.

**Comment by jeddy3** [comment]
> Is using the `--custom-formatter` an option, pointing towards your own formatter that outputs the text in an ideal format for your parsing?

Yep, the `--custom-formatter` option was added precisely for this type of use case. This is the very first request for a formatter of this type, and so we should be hesitant about adding it. My recommendation would be to try to make overcommit compatible with the `--custom-formatter` tag.

**Comment by Ruffeng** [comment]
@jeddy3 --custom-formatter sounds great, although I am not sure if it's the best option to create a JS file in a ruby gem. As well I think would be hard to reference the file, depending on the system, a gem can be stored in one folder or another

**Comment by ntwb** [comment]
The [stylelint-formatter-compact](https://github.com/ntwb/stylelint-formatter-compact) formats similar to what you are after @Ruffeng, or you could fork it and tweak [index.js](https://github.com/ntwb/stylelint-formatter-compact/blob/master/index.js) to your specific requirements.

The output of the formatter is:

```shell
/Users/netweb/dev/my-project/css/file.css: line 1149, col 34, error - Unexpected empty block (block-no-empty)
/Users/netweb/dev/my-project/css/file.css: line 1157, col 34, error - Unexpected empty block (block-no-empty)
```

**Comment by jeddy3** [comment]
@ntwb Could we bundle the compact formatter in stylelint? It sounds like that would make @Ruffeng life easier.

**Comment by ntwb** [comment]
@jeddy3 @Ruffeng I've added the _compact_ formatter in #3488

**Comment by Ruffeng** [comment]
Thanks @ntwb @jeddy3. I will add it to overcommit!!

**Comment by Ruffeng** [comment]
@jeddy3 @ntwb just let you know I sent the [PR at overcommit's side](https://github.com/brigade/overcommit/pull/590).

Thanks for the help and support!
